profile2/impl/src/java/org/sakaiproject/profile2/logic/ProfileImageLogicImpl.java

```diff
@@ -977,21 +1049,85 @@ else if(StringUtils.equals(pattern, "ONE_DEEP")) {
 		} 
 		return data;
 	}
-	
-	
-	@Setter
-	private SakaiProxy sakaiProxy;
-	
-	@Setter
-	private ProfilePrivacyLogic privacyLogic;
-	
-	@Setter
-	private ProfileConnectionsLogic connectionsLogic;
-	
-	@Setter
-	private ProfilePreferencesLogic preferencesLogic;
-	
-	@Setter
-	private ProfileDao dao;
-	
+
+	@Override
+	public ProfileImage getProfileAvatarInitials(String userUuid) {
+		if(cache.containsKey(userUuid)) {
+			return (ProfileImage)cache.get(userUuid);
+
+		} else {
+			ProfileImage image = new ProfileImage();
+			BufferedImage bufferedImage = new BufferedImage(ProfileConstants.PROFILE_AVATAR_WIDTH, ProfileConstants.PROFILE_AVATAR_HEIGHT, BufferedImage.TYPE_INT_ARGB);
+
+			String displayName = sakaiProxy.getUserDisplayName(userUuid);
+			String[] names = displayName.split(" ");
+			String initials = "";
+			int fontSize;
+			int profileInitialsSize = Integer.parseInt(sakaiProxy.getServerConfigurationParameter("profile2.avatar.initials.size", "2"));
+			switch(profileInitialsSize){
+				case 1:
+					initials = Character.toString(names[0].charAt(0));
+					fontSize = Integer.parseInt(sakaiProxy.getServerConfigurationParameter("profile2.avatar.initials.font.size", ProfileConstants.DFLT_PROFILE_AVATAR_FONT_SIZE_1_CHAR));
+					break;
+				case 2:
+				default:
+					for (int i=0; i < names.length;i++) {
+						if (i > 1) break;
+						initials += Character.toString(names[i].charAt(0));
+					}
+					fontSize = Integer.parseInt(sakaiProxy.getServerConfigurationParameter("profile2.avatar.initials.font.size", ProfileConstants.DFLT_PROFILE_AVATAR_FONT_SIZE_2_CHAR));
+					break;
+			}
+			initials = initials.toUpperCase();
+
+			Graphics2D background = bufferedImage.createGraphics();
+			background.setPaint(Color.decode(this.getAvatarInitialsColor(displayName)));
+			background.fillRect(0, 0, ProfileConstants.PROFILE_AVATAR_WIDTH, ProfileConstants.PROFILE_AVATAR_HEIGHT);
+
+			String fontFamily = sakaiProxy.getServerConfigurationParameter("profile2.avatar.initials.font", ProfileConstants.DFLT_PROFILE_AVATAR_FONT_FAMILY);
+			Graphics2D initialsg2d = bufferedImage.createGraphics();
+			initialsg2d.setPaint(Color.WHITE);
+			initialsg2d.setFont(new Font(fontFamily, Font.PLAIN, fontSize));
+			FontMetrics fm = initialsg2d.getFontMetrics();
+			int x = (ProfileConstants.PROFILE_AVATAR_WIDTH/2) - fm.stringWidth(initials) / 2;
+			int y = (ProfileConstants.PROFILE_AVATAR_HEIGHT - fm.getHeight()) / 2 + fm.getAscent();
+			initialsg2d.setRenderingHint(RenderingHints.KEY_TEXT_ANTIALIASING, RenderingHints.VALUE_TEXT_ANTIALIAS_ON);
+			initialsg2d.drawString(initials, x, y);
+			initialsg2d.dispose();
+
+			ByteArrayOutputStream baos = new ByteArrayOutputStream();
+			byte[] bytes = null;
+			try {
+				ImageIO.write(bufferedImage, "png", baos);
+				baos.flush();
+				bytes = baos.toByteArray();
+				baos.close();
```

Comment:
close should be in a finally and you could maybe use the try with resources

Reply:
Hi @ern , would it be ok like this?

`			byte[] bytes = null;
			try (ByteArrayOutputStream baos = new ByteArrayOutputStream()){
				ImageIO.write(bufferedImage, "png", baos);
				bytes = baos.toByteArray();
			} catch (IOException ex) {
				log.error("Cannot generate profile avatar for the user {}", userUuid);
			}`